oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndex.java

```diff
@@ -765,6 +765,8 @@ private static void addNonFullTextConstraints(List<Query> qs,
             }
 
             if (isLike) {
+                // This code may be unreachable. Adding a log statement to try to identify usages of this code
+                LOG.info("Transforming 'like' constraint {} for restriction {}", first, pr);
```

Comment:
Was this usage identified from a test ? If yes, I think we can remove this from the PR ? If not - then I think trying to add a test here would be good.

Reply:
Ah - I just noticed this is LuceneIndex. This class is used for backward compatability of compatVersion 1 lucene index. 
I am not sure if there are any tests with lucene indexes of compatVersion 1 . But that could be used to hit this code. As per me, this should be lower priority.

Reply:
Thanks for the clarification that the LuceneIndex class is used for compatVersion 1, that explains why I could not hit this part of the code. 
Is compatVersion 1 still being used? As I assume the intention is to move away from this code, then I guess it's not worth to fix it. My suggestion is to remove the log statement (since we already know when this code is used) and leave a comment mentioning that the logic is wrong, but that we will not fix it.

Reply:
I don't think it's being used. This is where the compat version is being set https://github.com/apache/jackrabbit-oak/blob/trunk/oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/util/IndexDefinitionBuilder.java#L87. and it sets it to 2. 

I think this code is just for backward compatibility for any repositories that probably still use an older version of lucene.



>My suggestion is to remove the log statement (since we already know when this code is used) and leave a comment mentioning that the logic is wrong, but that we will not fix it.

+1